api/httpjson/RPCserver.go

```diff
@@ -292,9 +272,35 @@ func (s *RPCServer) Start() {
 	}
 
 	httpServer.SetKeepAlivesEnabled(config.Parameters.RPCKeepAlivesEnabled)
+	s.httpServer = httpServer
+}
 
-	go httpServer.Serve(listener)
-	go httpServer.Serve(tlsListener)
+func (s *RPCServer) StartHttp() {
+	listener, err := net.Listen("tcp", s.httpListener)
+	if err != nil {
+		log.Error("net.Listen: ", err.Error())
+		return
+	}
+	s.httpServer.Serve(listener)
+}
+
+func (s *RPCServer) StartTls(httpsCertReady chan struct{}) {
+	for {
+		timer := time.NewTimer(time.Second * 5)
+		select {
+		case <-httpsCertReady:
+			log.Info("https cert ready")
+			tlsListener, err := s.initTlsListen()
+			if err != nil {
+				log.Errorf("Https Cert: %v", err.Error())
+				return
+			}
+			//s.setRPCTlsInfo()
+			s.httpServer.Serve(tlsListener)
+		case <-timer.C:
```

Comment:
You don't have to change this but you can use `<-time.After(5*time.Second)` to be more succinct.